plugins/modules/smart_proxy.py

```diff
@@ -101,13 +111,49 @@ def main():
         foreman_spec=dict(
             name=dict(required=True),
             url=dict(required=True),
+            lifecycle_environments=dict(required=False, type='entity_list'),
             download_policy=dict(required=False, choices=['background', 'immediate', 'on_demand']),
         ),
-        required_plugins=[('katello', ['download_policy'])],
+        required_plugins=[('katello', ['lifecycle_environments', 'download_policy'])],
     )
 
     with module.api_connection():
-        module.run()
+        handle_lifecycle_environments = not module.desired_absent and 'lifecycle_environments' in module.foreman_params
+        if handle_lifecycle_environments:
+            module.lookup_entity('lifecycle_environments')
+
+        lifecycle_environments = module.foreman_params.pop('lifecycle_environments', [])
+        smart_proxy = module.run()
+
+        if handle_lifecycle_environments:
+            payload = {
+                'id': smart_proxy['id'],
+            }
+            current_lces = module.resource_action('capsule_content', 'lifecycle_environments', payload, record_change=False)
```

Comment:
There's still some thought needed around this in check_mode.

Reply:
So basically what we need is to give check mode the list of what _should_ happen, right?

I've been trying to make the code do that, but it doesn't seem to work, it still barfs with

```
TASK [assert] ******************************************************************
task path: /home/jeff/git/foreman-ansible-modules/tests/test_playbooks/tasks/smart_proxy.yml:17
fatal: [tests]: FAILED! => {
    "assertion": "result.changed == expected_change", 
    "changed": false, 
    "evaluated_to": false, 
    "msg": "Ensuring Smart Proxy is present failed! (expected_change: False)"

```

Reply:
So let's try to get `check_mode` work. I think it should be:
```suggestion
            if smart_proxy['id'] == -1:
                # just created a fake entity, so fake the environments too
                current_lces = {'results': []}
            else:
                current_lces = module.resource_action('capsule_content', 'lifecycle_environments', payload, ignore_check_mode=True, record_change=False)
```

Reply:
Thanks a lot! That was the fix!

I was trying with ```module.check_mode``` and without the ```ignore_check_mode``` in the resource action :-)